tests/js/server/paths/filesystem.js

```diff
@@ -0,0 +1,350 @@
+/*jshint globalstrict:false, strict:false */
+/* global fail, getOptions, assertTrue, assertFalse, assertEqual, assertNotEqual, assertUndefined */
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief teardown for dump/reload tests
+///
+/// @file
+///
+/// DISCLAIMER
+///
+/// Copyright 2010-2012 triagens GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB Inc, Cologne, Germany
+///
+/// @author Wilfried Goesgens
+/// @author Copyright 2019, ArangoDB Inc, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+const fs = require('fs');
+const internal = require('internal');
+
+let rootDir = fs.join(fs.getTempPath(), '..');
+let testFilesDir = rootDir;
+
+
+const topLevelAllowed = fs.join(testFilesDir, 'allowed');
```

Comment:
I would have expected the explicit usage of Unicode paths _here_.

Reply:
Whe whole suite works in a Unicode path, so everything containing the root directory as prefix is tested with unicode strings. Do we gain anything by making it explicit?